src/frontend/js/pages/TeacherDashboardContractsLoader/TeacherDashboardContracts/index.spec.tsx

```diff
@@ -20,129 +23,273 @@ jest.mock('utils/context', () => ({
   }).one(),
 }));
 
-const Wrapper = ({ children }: PropsWithChildren) => {
+jest.mock('components/ContractFrame', () => ({
+  __esModule: true,
+  OrganizationContractFrame: ({
+    isOpen,
+    organizationId,
+  }: {
+    isOpen: boolean;
+    organizationId: string;
+  }) => isOpen && <p>ContractFrame opened for {organizationId}</p>,
+}));
+
+const Wrapper = ({ path, initialEntry }: { path: string; initialEntry: string }) => {
   return (
     <IntlProvider locale="en">
       <QueryClientProvider client={createTestQueryClient({ user: true })}>
         <JoanieSessionProvider>
-          <CunninghamProvider>{children}</CunninghamProvider>
+          <CunninghamProvider>
+            <MemoryRouter initialEntries={[initialEntry]}>
+              <Routes>
+                <Route path={path} element={<TeacherDashboardContracts />} />
+              </Routes>
+            </MemoryRouter>
+          </CunninghamProvider>
         </JoanieSessionProvider>
       </QueryClientProvider>
     </IntlProvider>
   );
 };
 
 describe('pages/TeacherDashboardContracts', () => {
```

Comment:
we could maybe add a test for empty list

Reply:
Right !